new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java

```diff
@@ -0,0 +1,313 @@
+      String kindRepoImage = KIND_REPO + wlsBaseImage.substring(TestConstants.OCR_REGISTRY.length() + 1);
+      logger.info("Using image {0}", kindRepoImage);
+      wlsBaseImage = kindRepoImage;
+      isUseSecret = false;
+    } else {
+      // create pull secrets for WebLogic image when running in non Kind Kubernetes cluster
+      createOCRRepoSecret(domainNamespace);
+    }
+
+    // create the domain CR with a pre-defined configmap
+    createDomainResource(domainUid, domainNamespace, adminSecretName,
+        REPO_SECRET_NAME, encryptionSecretName,
+        replicaCount, configMapName);
+
+    // wait for the domain to exist
+    logger.info("Check for domain custom resource in namespace {0}", domainNamespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for domain {0} to be created in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                domainUid,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainExists(domainUid, DOMAIN_VERSION, domainNamespace));
+  }
+
+  /**
+   * Verify all server pods are running.
+   * Verify all k8s services for all servers are created.
+   */
+  @BeforeEach
+  public void beforeEach() {
+    logger.info("Check admin service and pod {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace);
+    // check managed server services and pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server services and pods are created in namespace {0}",
+          domainNamespace);
+      checkPodReadyAndServiceExists(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+  }
+
+  /**
+   * Create a WebLogic domain with SSL enabled in WebLogic configuration.
+   * Create a domain CRD with a channel with the name `default-secure`.
+   * Make sure an external NodePort service is created in domain namespace.
+   * Make sure WebLogic console is accessible through the `default-secure` service. 
+   * Make sure WebLogic console is accessible through the `default` service.  
+   */
+  @Test
+  @DisplayName("Verify the secure service through SSL port")
+  public void testVerifySecurePortService() {
+    int sslNodePort = getServiceNodePort(
+         domainNamespace, adminServerPodName + "-external", "default-secure");
+    logger.info("Found the administration service nodePort {0}", sslNodePort);
+    String curlCmd = "curl -sk --show-error --noproxy '*' "
+        + " https://" + K8S_NODEPORT_HOST + ":" + sslNodePort
+        + "/console/login/LoginForm.jsp --write-out %{http_code} -o /dev/null";
+    logger.info("Executing defaukt-admin nodeport curl command {0}", curlCmd);
+    assertTrue(callWebAppAndWaitTillReady(curlCmd, 10));
```

Comment:
same comment as the other test class

Reply:
These are boiler plate generic code from existing Mii Domain Class.  I am not building any domain images in this class. The ImageBuilder class will build the domain image by default.